java-frontend/src/main/java/org/sonar/java/se/xproc/BehaviorCache.java

```diff
@@ -91,6 +110,12 @@ private static boolean isObjectsRequireNonNullMethod(Symbol symbol) {
     return symbol.owner().type().is("java.util.Objects") && "requireNonNull".equals(symbol.name());
   }
 
+  private static boolean isLog4jOrSpringAssertNotNull(Symbol symbol) {
+    Type ownerType = symbol.owner().type();
+    return (ownerType.is("org.apache.logging.log4j.core.util.Assert") && "requireNonNull".equals(symbol.name()))
+      || (ownerType.is("org.springframework.util.Assert") && "notNull".equals(symbol.name()));
```

Comment:
spring framework `Assert` class also have methods `isNull()` and `notEmpty()` which should be covered as well in my opinion.